projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -146,14 +146,39 @@ public function get_track_list() {
 	 * @return string Plain text string.
 	 */
 	protected function get_plain_text( $str ) {
+		return $this->sanitize_and_decode_text( $str, true );
+	}
+
+	/**
+	 * Formats strings as safe HTML.
+	 *
+	 * @param string $str Input string.
+	 * @return string HTML text string safe for post_content.
+	 */
+	protected function get_html_text( $str ) {
+		return $this->sanitize_and_decode_text( $str, false );
+	}
+
+	/**
+	 * Strip unallowed html tags and decode entities.
+	 *
+	 * @param string  $str Input string.
+	 * @param boolean $strip_all_tags Strip all tags, otherwise allow post_content safe tags.
+	 * @return string Sanitized and decoded text.
+	 */
+	protected function sanitize_and_decode_text( $str, $strip_all_tags = true ) {
```

Comment:
💯

Reply:
in the future, this helper could accept a tags array as a parameter too. If so, we use `wp_kses`:

```php
protected function sanitize_and_decode_text( $str, $strip_all_tags = true ) {
	// ...

	if ( is_bool( $strip_all_tags ) && $strip_all_tags === true ) {
		// Make sure there are no tags.
		$str = wp_strip_all_tags( $str );
	} elseif ( is_array( $strip_all_tags ) && ! empty( $strip_all_tags ) {
		$str = wp_kses( $str, strip_all_tags );
	} else {
		$str = wp_kses_post( $str );
	}
	// ...
}
```